[WPE][GTK] Move network session APIs to a new WebKitNetworkSession class

carlosgcampos opened the pull request:
#### 2d1b53b211b43047b94b7d2ef8a3907f00e724fb
<pre>
[WPE][GTK] Move network session APIs to a new WebKitNetworkSession class
<a href="https://bugs.webkit.org/show_bug.cgi?id=222366">https://bugs.webkit.org/show_bug.cgi?id=222366</a>

Reviewed by Michael Catanzaro.

Add new WebKitNetworkSession class to expose all the networking APIs
that were part of WebKitWebContext and WebKitWebsiteDataManager. Also
switch to use a single global network process.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/IconDatabase.cpp:
(WebKit::IconDatabase::IconDatabase):
* Source/WebKit/UIProcess/API/glib/WebKitAutocleanups.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitAutomationSession.cpp:
(webkitAutomationSessionCreate):
* Source/WebKit/UIProcess/API/glib/WebKitNavigationClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp: Added.
(_WebKitNetworkSessionPrivate::_WebKitNetworkSessionPrivate):
(webkitNetworkSessionGetProperty):
(webkitNetworkSessionSetProperty):
(webkitNetworkSessionConstructed):
(webkit_network_session_class_init):
(webkitNetworkSessionDownloadStarted):
(createDefaultNetworkSession):
(webkit_network_session_get_default):
(webkit_network_session_new):
(webkit_network_session_new_ephemeral):
(webkit_network_session_is_ephemeral):
(webkit_network_session_get_website_data_manager):
(webkit_network_session_get_cookie_manager):
(webkit_network_session_set_itp_enabled):
(webkit_network_session_get_itp_enabled):
(webkit_network_session_set_persistent_credential_storage_enabled):
(webkit_network_session_get_persistent_credential_storage_enabled):
(webkit_network_session_set_tls_errors_policy):
(webkit_network_session_get_tls_errors_policy):
(webkit_network_session_allow_tls_certificate_for_host):
(webkit_network_session_set_proxy_settings):
(webkit_network_session_set_memory_pressure_settings):
(webkit_network_session_get_itp_summary):
(webkit_network_session_get_itp_summary_finish):
(webkit_network_session_prefetch_dns):
(webkit_network_session_download_uri):
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.h.in: Added.
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSessionPrivate.h: Copied from Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManagerPrivate.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
(webkitWebContextGetNetworkSessionForAutomation):
(webkitWebContextGetProperty):
(webkitWebContextSetProperty):
(webkitWebContextConstructed):
(webkitWebContextDispose):
(webkit_web_context_class_init):
(webkit_web_context_set_network_proxy_settings):
(webkit_web_context_prefetch_dns):
(webkitWebContextCreatePageForWebView):
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebContextPrivate.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewIsEphemeral):
(webkitWebViewGetFaviconDatabase):
(webkitWebViewRequestFavicon):
(webkitWebViewWatchForChangesInFavicon):
(webkitWebViewConstructed):
(webkitWebViewSetProperty):
(webkitWebViewGetProperty):
(webkitWebViewDispose):
(webkit_web_view_class_init):
(webkitWebViewLoadChanged):
(webkitWebViewLoadFailedWithTLSErrors):
(webkitWebViewGetLoadDecisionForIcon):
(webkitWebViewSetIcon):
(webkitWebViewHandleAuthenticationChallenge):
(webkitWebViewGetWebsiteDataManager):
(webkit_web_view_get_network_session):
(webkitWebViewDisconnectFaviconDatabaseSignalHandlers): Deleted.
(webkit_web_view_download_uri): Deleted.
(webkit_web_view_get_tls_info): Deleted.
(webkit_web_view_get_snapshot): Deleted.
(webkit_web_view_get_snapshot_finish): Deleted.
(webkitWebViewWebProcessTerminated): Deleted.
(webkit_web_view_is_editable): Deleted.
(webkit_web_view_set_editable): Deleted.
(webkit_web_view_get_editor_state): Deleted.
(webkit_web_view_get_session_state): Deleted.
(webkit_web_view_restore_session_state): Deleted.
(webkit_web_view_add_frame_displayed_callback): Deleted.
(webkit_web_view_remove_frame_displayed_callback): Deleted.
(webkit_web_view_send_message_to_page): Deleted.
(webkit_web_view_send_message_to_page_finish): Deleted.
(webkit_web_view_set_input_method_context): Deleted.
(webkit_web_view_get_input_method_context): Deleted.
(webkit_web_view_get_website_policies): Deleted.
(webkitWebViewSetIsWebProcessResponsive): Deleted.
(webkit_web_view_get_is_web_process_responsive): Deleted.
(webkit_web_view_terminate_web_process): Deleted.
(webkit_web_view_set_cors_allowlist): Deleted.
(webkitWebViewConfigureMediaCapture): Deleted.
(webkit_web_view_get_camera_capture_state): Deleted.
(webkit_web_view_set_camera_capture_state): Deleted.
(webkit_web_view_get_microphone_capture_state): Deleted.
(webkit_web_view_set_microphone_capture_state): Deleted.
(webkit_web_view_get_display_capture_state): Deleted.
(webkit_web_view_set_display_capture_state): Deleted.
(webkitWebViewForceRepaintForTesting): Deleted.
(webkitSetCachedProcessSuspensionDelayForTesting): Deleted.
(webkit_web_view_get_web_extension_mode): Deleted.
(webkit_web_view_get_default_content_security_policy): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp:
(webkitWebsiteDataManagerConstructed):
(webkit_website_data_manager_class_init):
(webkitWebsiteDataManagerGetDataStore):
(webkitWebsiteDataManagerCreate):
(webkit_website_data_manager_get_dom_cache_directory):
(webkitWebsiteDataManagerGetFaviconDatabasePath):
(webkit_website_data_manager_set_favicons_enabled):
(webkit_website_data_manager_get_favicons_enabled):
(webkit_website_data_manager_get_favicon_database):
(webkitITPThirdPartyCreate):
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManagerPrivate.h:
* Source/WebKit/UIProcess/API/glib/webkit.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewGtk.cpp:
(webkit_web_view_new_with_context):
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp:
(webkit_web_view_new_with_context):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::networkProcessForSession):
* Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
* Tools/MiniBrowser/gtk/BrowserTab.c:
(tlsErrorsDialogResponse):
* Tools/MiniBrowser/gtk/BrowserWindow.c:
(webViewTitleChanged):
(downloadStarted):
(webViewCreate):
(webViewDecidePolicy):
(newTabCallback):
(openPrivateWindow):
(browserWindowFinalize):
(browser_window_new):
(browser_window_get_network_session):
(browser_window_load_session):
(browser_window_get_or_create_web_view_for_automation):
* Tools/MiniBrowser/gtk/BrowserWindow.h:
* Tools/MiniBrowser/gtk/main.c:
(createBrowserTab):
(aboutDataScriptMessageReceivedCallback):
(aboutDataHandleRequest):
(aboutITPHandleRequest):
(aboutURISchemeRequestCallback):
(activate):
* Tools/MiniBrowser/wpe/main.cpp:
(activate):
(main):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestAuthentication.cpp:
(testWebViewAuthenticationStorage):
(ProxyAuthenticationTest::ProxyAuthenticationTest):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestAutomationSession.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestCookieManager.cpp:
(testCookieManagerAcceptPolicy):
(testCookieManagerEphemeral):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestDownloads.cpp:
(testDownloadEphemeralContext):
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestSSL.cpp:
(testSSL):
(testInsecureContent):
(testTLSErrorsPolicy):
(testTLSErrorsRedirect):
(testTLSErrorsHTTPAuth):
(testLoadFailedWithTLSErrors):
(testSubresourceLoadFailedWithTLSErrors):
(testWebSocketTLSErrors):
(testTLSErrorsEphemeral):
(testClientSideCertificate):
(testWebSocketClientSideCertificate):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitFaviconDatabase.cpp:
(testFaviconDatabaseInitialization):
(testFaviconDatabaseGetFavicon):
(testFaviconDatabaseEphemeral):
(testFaviconDatabaseClear):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitNetworkSession.cpp: Added.
(testNetworkSessionDefault):
(testNetworkSessionEphemeral):
(serverCallback):
(ProxyTest::webSocketProxyServerCallback):
(ProxyTest::ProxyTest):
(ProxyTest::loadURIAndGetMainResourceData):
(ProxyTest::proxyServerPortAsString):
(ProxyTest::webSocketConnected):
(ProxyTest::createWebSocketAndWaitUntilConnected):
(webSocketServerCallback):
(ephemeralViewloadChanged):
(testNetworkSessionProxySettings):
(beforeAll):
(afterAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp:
(testWebViewEphemeral):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp:
(WebsiteDataTest::WebsiteDataTest):
(testWebsiteDataConfiguration):
(testWebsiteDataEphemeral):
(testWebsiteDataITP):
(MemoryPressureTest::setup):
(MemoryPressureTest::teardown):
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestAutocleanups.cpp:
(testUIProcessAutocleanups):
* Tools/TestWebKitAPI/glib/CMakeLists.txt:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.cpp:
(main):
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::Test):
(Test::~Test):
* Tools/TestWebKitAPI/glib/WebKitGLib/WebViewTest.cpp:
(WebViewTest::initializeWebView):

Canonical link: <a href="https://commits.webkit.org/259433@main">https://commits.webkit.org/259433@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/649d4a75faede98ce0709319dcad0c8444997846

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104878 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13958 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37773 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114146 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/174338 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108785 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/15089 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4882 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97205 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/113169 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110638 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11659 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94669 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39178 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93527 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26288 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80840 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7302 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27648 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7397 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4235 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13453 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47198 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9187 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3455 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp:
```diff
@@ -0,0 +1,657 @@
+    PAL::HysteresisActivity dnsPrefetchHystereris;
+};
+
+WEBKIT_DEFINE_FINAL_TYPE(WebKitNetworkSession, webkit_network_session, G_TYPE_OBJECT)
```
```suggestion
WEBKIT_DEFINE_FINAL_TYPE_IN_2022_API(WebKitNetworkSession, webkit_network_session, G_TYPE_OBJECT)
```
```suggestion
WEBKIT_DEFINE_FINAL_TYPE(WebKitNetworkSession, webkit_network_session, G_TYPE_OBJECT)
```

carlosgcampos replied:
This is 2022 API only, I guess we don't need the IN_2022_API macro, no?

aperezdc replied:
@carlosgcampos Yes, you can use `WEBKIT_DEFINE_FINAL_TYPE` directly in this case.

mcatanzaro commented on Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
```diff
@@ -66,3 +66,18 @@ accordingly.
+WebKit now uses a single global network process for all web contexts, and different
+network sessions can be created and used in the same network process. All the networking
+APIs have been moved from WebKitWebContext and WebKitWebsiteDataManager to the new class
+WebKitNetworkSession. There's a default global persistent session that can be get with
```
```suggestion
[type@WebKit.NetworkSession]. There's a default global persistent session that you can get with
```

mcatanzaro commented on Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
```diff
@@ -66,3 +66,18 @@ accordingly.
+WebKitNetworkSession. There's a default global persistent session that can be get with
+webkit_network_session_get_default(). You can also create new sessions with
+webkit_network_session_new() for persistent sessions and webkit_network_session_new_ephemeral()
+for ephemeral sessions. It's no longer possible to create a WebKitWebsiteDataManager, it's now
```
```suggestion
for ephemeral sessions. It's no longer possible to create a [type@WebKit.WebsiteDataManager]; it's now
```

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
```diff
@@ -2702,9 +2795,13 @@ void webkitWebViewSubmitFormRequest(WebKitWebView* webView, WebKitFormSubmission
 void webkitWebViewHandleAuthenticationChallenge(WebKitWebView* webView, AuthenticationChallengeProxy* authenticationChallenge)
 {
+#if ENABLE(2022_GLIB_API)
+    bool credenialStorageEnabled = webkit_network_session_get_persistent_credential_storage_enabled(webView->priv->networkSession.get());
```
```suggestion
    bool credentialStorageEnabled = webkit_network_session_get_persistent_credential_storage_enabled(webView->priv->networkSession.get());
```

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
```diff
@@ -2702,9 +2795,13 @@ void webkitWebViewSubmitFormRequest(WebKitWebView* webView, WebKitFormSubmission
     auto* websiteDataManager = webkit_web_view_get_website_data_manager(webView);
-    webView->priv->authenticationRequest = adoptGRef(webkitAuthenticationRequestCreate(authenticationChallenge,
-        webView->priv->isEphemeral, webkit_website_data_manager_get_persistent_credential_storage_enabled(websiteDataManager)));
+    bool credenialStorageEnabled = webkit_website_data_manager_get_persistent_credential_storage_enabled(websiteDataManager);
```
```suggestion
    bool credentialStorageEnabled = webkit_website_data_manager_get_persistent_credential_storage_enabled(websiteDataManager);
```

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
```diff
@@ -2702,9 +2795,13 @@ void webkitWebViewSubmitFormRequest(WebKitWebView* webView, WebKitFormSubmission
-        webView->priv->isEphemeral, webkit_website_data_manager_get_persistent_credential_storage_enabled(websiteDataManager)));
+    bool credenialStorageEnabled = webkit_website_data_manager_get_persistent_credential_storage_enabled(websiteDataManager);
+#endif
+    webView->priv->authenticationRequest = adoptGRef(webkitAuthenticationRequestCreate(authenticationChallenge, webkitWebViewIsEphemeral(webView), credenialStorageEnabled));
```
```suggestion
    webView->priv->authenticationRequest = adoptGRef(webkitAuthenticationRequestCreate(authenticationChallenge, webkitWebViewIsEphemeral(webView), credentialStorageEnabled));
```

mcatanzaro commented on Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
```diff
@@ -66,3 +66,18 @@ accordingly.
+
+WebKit now uses a single global network process for all web contexts, and different
+network sessions can be created and used in the same network process. All the networking
+APIs have been moved from WebKitWebContext and WebKitWebsiteDataManager to the new class
```
```suggestion
APIs have been moved from [type@WebKit.WebContext] and [type@WebKit.WebsiteDataManager] to the new class
```

mcatanzaro commented on Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
```diff
@@ -66,3 +66,18 @@ accordingly.
+network sessions can be created and used in the same network process. All the networking
+APIs have been moved from WebKitWebContext and WebKitWebsiteDataManager to the new class
+WebKitNetworkSession. There's a default global persistent session that can be get with
+webkit_network_session_get_default(). You can also create new sessions with
```
```suggestion
[func@WebKit.NetworkSession.get_default]. You can also create new sessions with
```

mcatanzaro commented on Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
```diff
@@ -66,3 +66,18 @@ accordingly.
+APIs have been moved from WebKitWebContext and WebKitWebsiteDataManager to the new class
+WebKitNetworkSession. There's a default global persistent session that can be get with
+webkit_network_session_get_default(). You can also create new sessions with
+webkit_network_session_new() for persistent sessions and webkit_network_session_new_ephemeral()
```
```suggestion
[ctor@WebKit.NetworkSession.new] for persistent sessions and [ctor@WebKit.NetworkSession.new_ephemeral]
```

mcatanzaro commented on Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
```diff
@@ -66,3 +66,18 @@ accordingly.
+for ephemeral sessions. It's no longer possible to create a WebKitWebsiteDataManager, it's now
+created by the WebKitNetworkSession automatically at construction time. The WebKitNetworkSession
+to be used must be passed to the WebKitWebView as a constructor parameter. You can pass the
+same WebKitNetworkSession object to several web views to use the same session. The only exception
```
```suggestion
created by the [type@WebKit.NetworkSession] automatically at construction time. The [type@WebKit.NetworkSession]
to be used must be passed to the [type@WebKit.WebView] as a construct parameter. You can pass the
same [type@WebKit.NetworkSession] object to several web views to use the same session. The only exception
```

mcatanzaro reviewed (approved):
Wow, that was a lot of work.

carlosgcampos committed:
[WPE][GTK] Move network session APIs to a new WebKitNetworkSession class
https://bugs.webkit.org/show_bug.cgi?id=222366

Reviewed by Michael Catanzaro.

Add new WebKitNetworkSession class to expose all the networking APIs
that were part of WebKitWebContext and WebKitWebsiteDataManager. Also
switch to use a single global network process.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/IconDatabase.cpp:
(WebKit::IconDatabase::IconDatabase):
* Source/WebKit/UIProcess/API/glib/WebKitAutocleanups.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitAutomationSession.cpp:
(webkitAutomationSessionCreate):
* Source/WebKit/UIProcess/API/glib/WebKitNavigationClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp: Added.
(_WebKitNetworkSessionPrivate::_WebKitNetworkSessionPrivate):
(webkitNetworkSessionGetProperty):
(webkitNetworkSessionSetProperty):
(webkitNetworkSessionConstructed):
(webkit_network_session_class_init):
(webkitNetworkSessionDownloadStarted):
(createDefaultNetworkSession):
(webkit_network_session_get_default):
(webkit_network_session_new):
(webkit_network_session_new_ephemeral):
(webkit_network_session_is_ephemeral):
(webkit_network_session_get_website_data_manager):
(webkit_network_session_get_cookie_manager):
(webkit_network_session_set_itp_enabled):
(webkit_network_session_get_itp_enabled):
(webkit_network_session_set_persistent_credential_storage_enabled):
(webkit_network_session_get_persistent_credential_storage_enabled):
(webkit_network_session_set_tls_errors_policy):
(webkit_network_session_get_tls_errors_policy):
(webkit_network_session_allow_tls_certificate_for_host):
(webkit_network_session_set_proxy_settings):
(webkit_network_session_set_memory_pressure_settings):
(webkit_network_session_get_itp_summary):
(webkit_network_session_get_itp_summary_finish):
(webkit_network_session_prefetch_dns):
(webkit_network_session_download_uri):
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.h.in: Added.
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSessionPrivate.h: Copied from Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManagerPrivate.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:
(webkitWebContextGetNetworkSessionForAutomation):
(webkitWebContextGetProperty):
(webkitWebContextSetProperty):
(webkitWebContextConstructed):
(webkitWebContextDispose):
(webkit_web_context_class_init):
(webkit_web_context_set_network_proxy_settings):
(webkit_web_context_prefetch_dns):
(webkitWebContextCreatePageForWebView):
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebContextPrivate.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewIsEphemeral):
(webkitWebViewGetFaviconDatabase):
(webkitWebViewRequestFavicon):
(webkitWebViewWatchForChangesInFavicon):
(webkitWebViewConstructed):
(webkitWebViewSetProperty):
(webkitWebViewGetProperty):
(webkitWebViewDispose):
(webkit_web_view_class_init):
(webkitWebViewLoadChanged):
(webkitWebViewLoadFailedWithTLSErrors):
(webkitWebViewGetLoadDecisionForIcon):
(webkitWebViewSetIcon):
(webkitWebViewHandleAuthenticationChallenge):
(webkitWebViewGetWebsiteDataManager):
(webkit_web_view_get_network_session):
(webkitWebViewDisconnectFaviconDatabaseSignalHandlers): Deleted.
(webkit_web_view_download_uri): Deleted.
(webkit_web_view_get_tls_info): Deleted.
(webkit_web_view_get_snapshot): Deleted.
(webkit_web_view_get_snapshot_finish): Deleted.
(webkitWebViewWebProcessTerminated): Deleted.
(webkit_web_view_is_editable): Deleted.
(webkit_web_view_set_editable): Deleted.
(webkit_web_view_get_editor_state): Deleted.
(webkit_web_view_get_session_state): Deleted.
(webkit_web_view_restore_session_state): Deleted.
(webkit_web_view_add_frame_displayed_callback): Deleted.
(webkit_web_view_remove_frame_displayed_callback): Deleted.
(webkit_web_view_send_message_to_page): Deleted.
(webkit_web_view_send_message_to_page_finish): Deleted.
(webkit_web_view_set_input_method_context): Deleted.
(webkit_web_view_get_input_method_context): Deleted.
(webkit_web_view_get_website_policies): Deleted.
(webkitWebViewSetIsWebProcessResponsive): Deleted.
(webkit_web_view_get_is_web_process_responsive): Deleted.
(webkit_web_view_terminate_web_process): Deleted.
(webkit_web_view_set_cors_allowlist): Deleted.
(webkitWebViewConfigureMediaCapture): Deleted.
(webkit_web_view_get_camera_capture_state): Deleted.
(webkit_web_view_set_camera_capture_state): Deleted.
(webkit_web_view_get_microphone_capture_state): Deleted.
(webkit_web_view_set_microphone_capture_state): Deleted.
(webkit_web_view_get_display_capture_state): Deleted.
(webkit_web_view_set_display_capture_state): Deleted.
(webkitWebViewForceRepaintForTesting): Deleted.
(webkitSetCachedProcessSuspensionDelayForTesting): Deleted.
(webkit_web_view_get_web_extension_mode): Deleted.
(webkit_web_view_get_default_content_security_policy): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp:
(webkitWebsiteDataManagerConstructed):
(webkit_website_data_manager_class_init):
(webkitWebsiteDataManagerGetDataStore):
(webkitWebsiteDataManagerCreate):
(webkit_website_data_manager_get_dom_cache_directory):
(webkitWebsiteDataManagerGetFaviconDatabasePath):
(webkit_website_data_manager_set_favicons_enabled):
(webkit_website_data_manager_get_favicons_enabled):
(webkit_website_data_manager_get_favicon_database):
(webkitITPThirdPartyCreate):
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManagerPrivate.h:
* Source/WebKit/UIProcess/API/glib/webkit.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewGtk.cpp:
(webkit_web_view_new_with_context):
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp:
(webkit_web_view_new_with_context):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::networkProcessForSession):
* Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
* Tools/MiniBrowser/gtk/BrowserTab.c:
(tlsErrorsDialogResponse):
* Tools/MiniBrowser/gtk/BrowserWindow.c:
(webViewTitleChanged):
(downloadStarted):
(webViewCreate):
(webViewDecidePolicy):
(newTabCallback):
(openPrivateWindow):
(browserWindowFinalize):
(browser_window_new):
(browser_window_get_network_session):
(browser_window_load_session):
(browser_window_get_or_create_web_view_for_automation):
* Tools/MiniBrowser/gtk/BrowserWindow.h:
* Tools/MiniBrowser/gtk/main.c:
(createBrowserTab):
(aboutDataScriptMessageReceivedCallback):
(aboutDataHandleRequest):
(aboutITPHandleRequest):
(aboutURISchemeRequestCallback):
(activate):
* Tools/MiniBrowser/wpe/main.cpp:
(activate):
(main):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestAuthentication.cpp:
(testWebViewAuthenticationStorage):
(ProxyAuthenticationTest::ProxyAuthenticationTest):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestAutomationSession.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestCookieManager.cpp:
(testCookieManagerAcceptPolicy):
(testCookieManagerEphemeral):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestDownloads.cpp:
(testDownloadEphemeralContext):
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestSSL.cpp:
(testSSL):
(testInsecureContent):
(testTLSErrorsPolicy):
(testTLSErrorsRedirect):
(testTLSErrorsHTTPAuth):
(testLoadFailedWithTLSErrors):
(testSubresourceLoadFailedWithTLSErrors):
(testWebSocketTLSErrors):
(testTLSErrorsEphemeral):
(testClientSideCertificate):
(testWebSocketClientSideCertificate):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitFaviconDatabase.cpp:
(testFaviconDatabaseInitialization):
(testFaviconDatabaseGetFavicon):
(testFaviconDatabaseEphemeral):
(testFaviconDatabaseClear):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitNetworkSession.cpp: Added.
(testNetworkSessionDefault):
(testNetworkSessionEphemeral):
(serverCallback):
(ProxyTest::webSocketProxyServerCallback):
(ProxyTest::ProxyTest):
(ProxyTest::loadURIAndGetMainResourceData):
(ProxyTest::proxyServerPortAsString):
(ProxyTest::webSocketConnected):
(ProxyTest::createWebSocketAndWaitUntilConnected):
(webSocketServerCallback):
(ephemeralViewloadChanged):
(testNetworkSessionProxySettings):
(beforeAll):
(afterAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp:
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp:
(testWebViewEphemeral):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp:
(WebsiteDataTest::WebsiteDataTest):
(testWebsiteDataConfiguration):
(testWebsiteDataEphemeral):
(testWebsiteDataITP):
(MemoryPressureTest::setup):
(MemoryPressureTest::teardown):
(beforeAll):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestAutocleanups.cpp:
(testUIProcessAutocleanups):
* Tools/TestWebKitAPI/glib/CMakeLists.txt:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.cpp:
(main):
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.h:
(Test::Test):
(Test::~Test):
* Tools/TestWebKitAPI/glib/WebKitGLib/WebViewTest.cpp:
(WebViewTest::initializeWebView):

Canonical link: https://commits.webkit.org/259433@main
